src/resolvers/dockerfileTemplate.ts

```diff
@@ -0,0 +1,65 @@
+import * as Promise from 'bluebird'
+
+import * as DockerfileTemplate from 'dockerfile-template'
+
+import { Bundle, FileInfo, Resolver } from '../resolver'
+
+export default class DockerfileTemplateResolver implements Resolver {
+	public priority = 2
+	public name = 'Dockerfile.template'
+
+	private hasDockerfileTemplate = false
+	private templateContent: string
+	private bundle: Bundle
+
+	public entry(file: FileInfo): void {
+		if (file.name === 'Dockerfile.template') {
+			this.templateContent = file.contents
+			this.hasDockerfileTemplate = true
+		}
+	}
+
+	public provideBundle(bundle: Bundle): void {
+		this.bundle = bundle
+	}
+
+	public isSatisfied(): boolean {
+		return this.hasDockerfileTemplate
+	}
+
+	public resolve(): Promise<FileInfo[]> {
+		if (!this.hasDockerfileTemplate) {
+			throw new Error('Resolve called on resolver which is not satisfies')
+		}
```

Comment:
We don't do checks like this on the others, and I don't think we should bother, we should trust the guarantees the interface gives us.